### ISSUE TITLE
update user profile - confusing message about validateFields() fail

### ISSUE THREAD

**Issue by Jimmi08** [e107inc/e107#4564]
## Bug Description
not able to update (resave) usersettings 

```
Error
06 - Login Name - Duplicate value: admin
06 - Email address - Duplicate value: localhost@e107sk.com
```

Value after those lines:
```
// Now validate everything - just check everything that's been entered
$allData = validatorClass::validateFields($_POST,$userMethods->userVettingInfo, TRUE);		// Do basic validation
validatorClass::dbValidateArray($allData, $userMethods->userVettingInfo, 'user', $inp);		// Do basic DB-related checks
$userMethods->userValidation($allData);									// Do user-specific DB checks
print_a($allData);
```

It is not a clean installation, but I installed all not core plugins and set the bootstrap3 theme. 

```
Array
(
    [data] => Array
        (
            [user_image] => 
            [user_signature] => [html] ***[/html]
            [user_hideemail] => 0
        )

    [failed] => Array
        (
            [user_password] => 
            [user_loginname] => admin
            [user_email] => localhost@e107sk.com
        )

    [errors] => Array
        (
            [user_password] => 01
            [user_loginname] => 06
            [user_email] => 06
        )

)
```
 
## How to Reproduce
1. latest sync
2. resave your profile
 
## Expected Behavior
no error, it is not a new account. There is only a second test user with a different name and email.

## Screenshots

![image](https://user-images.githubusercontent.com/5429548/132981250-6523a700-362f-499f-a5e7-a4dc667ca255.png)

### PHP Version

```
PHP Version 7.4
```

**Comment by Jimmi08** [comment]
It looks like changes are saved.  I changed the title. It can be closed if this is the intention.

**Comment by Moc** [comment]
Confirmed, I get the same on my localhost dev installation.

**Comment by Deltik** [comment]
This is a regression caused by https://github.com/e107inc/e107/commit/20882920a0b68937570264949512acc0c4841dbd

Git bisect log:
```
www-data@e107dev:~/html/e107$ git bisect log
git bisect start
# bad: [036b301c31da28a7921137de808a5701361844e6] Merge pull request #4504 from Jimmi08/patch-27
git bisect bad 036b301c31da28a7921137de808a5701361844e6
# good: [a9c2ae3823e804657aff7a1c155a2f39152f4229] Fix #4405: PHP 8 compatibility: redirection::checkMembersOnly()
git bisect good a9c2ae3823e804657aff7a1c155a2f39152f4229
# good: [b994dd916dc074f61e5da95af327c135fbcb02a1] Bootstrap upgraded to v5.1.0
git bisect good b994dd916dc074f61e5da95af327c135fbcb02a1
# good: [f5bb80607a304ea8dc2270b7600140a0fe3c4834] word limit per record added.
git bisect good f5bb80607a304ea8dc2270b7600140a0fe3c4834
# bad: [3e52f29087b1efff0cedb41dd9a2352eeb5470d5] Merge pull request #4554 from Deltik/php-8.1
git bisect bad 3e52f29087b1efff0cedb41dd9a2352eeb5470d5
# good: [834c713eef7b6739a4435e9c9cd9345e7ca4664a] Fix for e107_media and removal of old files.
git bisect good 834c713eef7b6739a4435e9c9cd9345e7ca4664a
# bad: [20882920a0b68937570264949512acc0c4841dbd] Fix all PHP 8.1 test failures
git bisect bad 20882920a0b68937570264949512acc0c4841dbd
# good: [64cd796605509181bbd871073fbf9f1f202b0d39] Update test dependencies
git bisect good 64cd796605509181bbd871073fbf9f1f202b0d39
# first bad commit: [20882920a0b68937570264949512acc0c4841dbd] Fix all PHP 8.1 test failures
```
